File: protocol/src/test/php/zfoophp/packet/EmptyObject.php
<?php

namespace zfoophp;


class EmptyObject {
    
}

class EmptyObjectRegistration implements IProtocolRegistration {
    public function protocolId(): int
    {
        return 0;
    }

    public function write(ByteBuffer $buffer, mixed $packet): void
    {
        if ($packet == null)
        {
            $buffer->writeInt(0);
            return;
        }
        $buffer->writeInt(-1);
    }

    public function read(ByteBuffer $buffer): mixed
    {
        $length = $buffer->readInt();
        $packet = new EmptyObject();
        if ($length == 0) {
            return $packet;
        }
        $beforeReadIndex = $buffer->getReadOffset();
        
        if ($length > 0) {
            $buffer->setReadOffset($beforeReadIndex + $length);
        }
        return $packet;
    }
}

File: protocol/src/test/ruby/zfooruby/packet/EmptyObject.rb

class EmptyObject
    
    def initialize()
        
    end
end

class EmptyObjectRegistration
    def protocolId()
        return 0
    end

    def write(buffer, packet)
        if packet.nil?
            buffer.writeInt(0)
            return
        end
        buffer.writeInt(-1)
    end

    def read(buffer)
        length = buffer.readInt()
        if length == 0
            return nil
        end
        beforeReadIndex = buffer.getReadOffset()
        packet = EmptyObject.new()
        
        if length > 0
            buffer.setReadOffset(beforeReadIndex + length)
        end
        return packet
    end
end